c#-mini/.md

```diff
@@ -0,0 +1,140 @@
+
+## Top-level statements and Implicit usings :rocket:
+
+Top-level statements are another new feature in C# 9. That allowed us to have a single file starting straight from the code statements, without a need of a class of method. This file serves as a project entry point. It's a great way to quickly prototype something or write a script. It's also a great way to get rid of the boilerplate code. C# 10 further simplified this with Implicit usings. Let's compare the old and new ways of writing a simple "Hello World" program:
+
+```csharp
+// Old way
+using System;
+
+namespace CSharp10
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Hello World!");
+        }
+    }
+}
+```
+
+```csharp
+// New way
+Console.WriteLine("Hello World!");
+```
+
+As you may see, that is a huge improvement. And it's not only about the number of lines. It's also about the number of concepts. The old way required us to understand namespaces, classes, methods, and their modifiers. The new way is much more straightforward.
+
+## A note about namespaces :alien:
+
+That's important to know that with top level statements you can have an program class. It's needed for example for ASP.NET Integration testing. This is done by placing the following code after the top-level statements:
+
+```csharp
+public partial class Program {}
+```
+
+But what namespace is it in? Is it somehow inferred from the project name? Turns out it's not in any namespace! Wait... Is it legal? Surprisingly enough, it is! What's even more shocking is that it was legal virtually from the beginning of C#! And without a namespace, the class is just visible without any `using`.
+
+What I still don't understand is: Why isn't it common practice to have classes without namespaces? I get why you'll want to skip, this for a public API. But here's an extensive list of places where you may use it:
+- **ASP.NET Controllers** - you hardly will reference it somewhere. Why would you need a namespaces
+- **Tests** - We can just group them by project name. Why declare a namespace?
+- **Domain Layer** - Let's be honest the layer most likely will be used in a single solution where it's unlikely to have classes with the same name. So why overhead!
+
+Unfortunately, I don't see a lot of discussion about the justification of having namespaces for every class in C#. So, maybe the article will be the spark.
+
+## Kill the .csproj? :fire:
+
+Top-level statements are a great leap towards making C# great for scripting and prototyping. But there's still a long way to go. And I feel like the main blocker is the necessity for a **csproj** file for any C# program. So, let's fantasize about how we can get rid of it in this post. I'll focus on solutions where we persist all the information csproj files contain but in the project's entry point file - `Program.cs` file instead. Here are my two ideas:
```

Comment:
I'll focus on solutions where we persist all the information a `.csproj` file contains but move it to a `.cs` file